tests/test_forms.py

```diff
@@ -725,6 +725,58 @@ def validate_some_slot(self, value, dispatcher, tracker, domain):
     assert events == [SlotSet("some_slot", "validated_value")]
 
 
+def test_validate_prefilled_slots():
+    # noinspection PyAbstractClass
+    class CustomFormAction(FormAction):
+        def name(self):
+            return "some_form"
+
+        @staticmethod
+        def required_slots(_tracker):
+            return ["some_slot", "some_other_slot"]
+
+        def validate_some_slot(self, value, dispatcher, tracker, domain):
+            if value == "some_value":
+                return {"some_slot": "validated_value"}
+            else:
+                return {"some_slot": None}
+
+    form = CustomFormAction()
+
+    tracker = Tracker(
+        "default",
+        {"some_slot": "some_value", "some_other_slot": "some_other_value"},
+        {
+            "entities": [{"entity": "some_slot", "value": "some_bad_value"}],
+            "text": "some text",
+        },
+        [],
+        False,
+        None,
+        {},
+        "action_listen",
+    )
+
+    events = form._activate_if_required(dispatcher=None, tracker=tracker, domain=None)
+    # check that the form was activated and prefilled slots were validated
+    assert events == [
+        Form("some_form"),
+        SlotSet("some_slot", "validated_value"),
+        SlotSet("some_other_slot", "some_other_value"),
+    ]
+
+    events.extend(
+        form._validate_if_required(dispatcher=None, tracker=tracker, domain=None)
+    )
+    # check that entities picked up in input overwrite prefilled slots
+    assert events == [
```

Comment:
@Ghostvv here i added an extra check because i wanted to make sure that we had ideal behavior:
If `some_slot` is pre-filled, we validate the value of whatever is in the slot during activation. If we also picked up the entity `some_slot` from the input, that takes precedence and gets validated too. In this case, even though the value pre-filled slot was valid, the user provided a non-valid entity in the input, so the slot was set to none. I think this makes sense because if the user is trying to input information for that entity, then it should take whatever they input and not whatever was in the pre-filled slot. Does that make sense to you?

Reply:
I suppose the only issue is that in order for it to overwrite, the timestamp of the event has to be later (pytest doesn't put timestamps on it). when do the timestamps actually get into the events -- will an event returned by the activate function correctly have a timestamp earlier than one returned by the validate function?

Reply:
I don't think timestamp is important here

Reply:
I just mean that the timestamp would be the one to decide if `some_slot` ends up getting set to `validated_value` or `None`, right?

Reply:
sorry, I'm not sure I understand what you mean, if there are prefilled slots, it will validate them, then if there are entities, it will call validate again, then in `tracker.events` they will be executed based on the order in the list of events returned by form, and since second validate is called later it should override previous changes